File: tests/solutions_js/two_sum.js
class Solution {
    twoSum(nums, target) {
        const lookup = {};
        for (let i = 0; i < nums.length; i++) {
            const complement = target - nums[i];
            if (complement in lookup) {
                return [lookup[complement], i];
            }
            lookup[nums[i]] = i;
        }
    }
}

File: tests/solutions_cpp/two_sum.cpp
#include "json_helper.h"

class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int, int> m;
        for (int i = 0; i < (int)nums.size(); i++) {
            int c = target - nums[i];
            auto it = m.find(c);
            if (it != m.end()) return {it->second, i};
            m[nums[i]] = i;
        }
        return {};
    }
};

int main() {
    auto args = parse_args();
    Solution sol;
    auto nums = to_vector_int(args[0]);
    int target = to_int(args[1]);
    auto result = sol.twoSum(nums, target);
    cout << to_json(result) << endl;
    return 0;
}
